dbt_platform_helper/utils/aws.py

```diff
@@ -20,62 +21,76 @@
 
 
 def get_aws_session_or_abort(aws_profile: str = None) -> boto3.session.Session:
-    aws_profile = aws_profile if aws_profile else os.getenv("AWS_PROFILE")
+    aws_profile = aws_profile or os.getenv("AWS_PROFILE")
     if aws_profile in AWS_SESSION_CACHE:
         return AWS_SESSION_CACHE[aws_profile]
 
-    # Check that the aws profile exists and is set.
-    click.secho(f"""Checking AWS connection for profile "{aws_profile}"...""", fg="cyan")
+    click.secho(f'Checking AWS connection for profile "{aws_profile}"...', fg="cyan")
 
     try:
         session = boto3.session.Session(profile_name=aws_profile)
+        sts = session.client("sts")
+        account_id, user_id = get_account_details(sts)
+        click.secho("Credentials are valid.", fg="green")
+
     except botocore.exceptions.ProfileNotFound:
-        click.secho(f"""AWS profile "{aws_profile}" is not configured.""", fg="red")
-        exit(1)
+        _handle_error(f'AWS profile "{aws_profile}" is not configured.')
     except botocore.exceptions.ClientError as e:
         if e.response["Error"]["Code"] == "ExpiredToken":
-            click.secho(
-                f"Credentials are NOT valid.  \nPlease login with: aws sso login --profile {aws_profile}",
-                fg="red",
+            _handle_error(
+                f"Credentials are NOT valid.  \nPlease login with: aws sso login --profile {aws_profile}"
             )
-            exit(1)
-
-    sts = session.client("sts")
-    try:
-        account_id, user_id = get_account_details(sts)
-        click.secho("Credentials are valid.", fg="green")
-    except (
-        botocore.exceptions.UnauthorizedSSOTokenError,
-        botocore.exceptions.TokenRetrievalError,
-        botocore.exceptions.SSOTokenLoadError,
-    ):
-        click.secho(
-            "The SSO session associated with this profile has expired or is otherwise invalid."
-            "To refresh this SSO session run `aws sso login` with the corresponding profile",
-            fg="red",
+    except botocore.exceptions.NoCredentialsError:
+        _handle_error(
+            "There are no credentials set for this session."
+            "To refresh this SSO session run `aws sso login` with the corresponding profile"
+        )
+    except botocore.exceptions.UnauthorizedSSOTokenError:
+        _handle_error(
+            "The SSO Token used for this session is unauthorised."
+            "To refresh this SSO session run `aws sso login` with the corresponding profile"
+        )
+    except botocore.exceptions.TokenRetrievalError:
+        _handle_error(
+            "Unable to retrieve the Token for this session."
+            "To refresh this SSO session run `aws sso login` with the corresponding profile"
```

Comment:
Might be worth extracting this string to a variable as it's repeated three times.